pkg/apis/monitoring/v1/types.go

```diff
@@ -615,6 +617,22 @@ type QueueConfig struct {
 	MaxBackoff string `json:"maxBackoff,omitempty"`
 }
 
+// Sigv4 optionally configures AWS's Signature Verification 4 signing process to
+// sign requests. Cannot be set at the same time as basic_auth or authorization.
+// +k8s:openapi-gen=true
+type Sigv4 struct {
+	// Region is the AWS region. If blank, the region from the default credentials chain used.
+	Region string `json:"region,omitempty"`
+	// AccessKey is the AWS API key. If blank, the environment variable `AWS_ACCESS_KEY_ID` is used
+	AccessKey string `json:"accessKey,omitempty"`
+	// SecretKey is the AWS API secret. If blank, the environment variable `AWS_SECRET_ACCESS_KEY` is used
+	SecretKey string `json:"secretKey,omitempty"`
```

Comment:
hm, both `AccessKey` and `SecretKey` should rather be of type `*v1.SecretKeySelector`.

Reply:
Question on this: can we reuse the Prometheus type `SigV4Config` rather than rolling our own type?
https://github.com/prometheus/prometheus/blob/d5c3c567d3e2badbdcc844edeaeff4e3b36c71dc/config/config.go#L742

Reply:
Edited the comment above to have a better understanding of the initial request.

Reply:
>  Does implies if they are stored in secrets, we should extract them from there.

yes you can have a look at GetSecretKey() for instance https://github.com/prometheus-operator/prometheus-operator/blob/63b3840645ce8ded070d1ceb7321ad19744a6644/pkg/assets/store.go#L263-L292